### PR TITLE
Small Refactoring: jefail refactored to C++

### PR DESCRIPTION
**Pull request by ahmadbeirkdar**
jefail refactored to C++, throws invalid arugment exception
Sending this PR mainly to see if throwing exceptions is fine, instead of writing to stderr then exiting. And the style of refactoring. 

```
96% tests passed, 15 tests failed out of 406

Total Test time (real) = 269.18 sec

The following tests FAILED:
	118 - g210a (SEGFAULT)
	123 - g221 (Failed)
	137 - g300 (Failed)
	157 - g331 (SEGFAULT)
	183 - g420stch (SEGFAULT)
	226 - g5x30 (Failed)
	234 - g602 (Failed)
	257 - g9x40 (Failed)
	272 - gebar (Failed)
	303 - gmmf (Failed)
	304 - gmmf1s (Failed)
	305 - gmmf1u (Failed)
	306 - gmmf1w (Failed)
	397 - gu (Failed)
	398 - gu0 (Failed)
Errors while running CTest
```

### CODE REVIEW

**Comment by ahmadbeirkdar** [comment]
Seems like linux is failing due to an old clang version being used to compiler, clang 7.0.0

File: jsrc/jeload.cpp [hunk 1]
```diff
@@ -169,7 +171,9 @@ int jefirst(int type,char* arg)
 	int r;
 	char* p;
 	char* q;
-	char* input=static_cast<char* >(malloc(2000+strlen(arg)));
+//	char* input=static_cast<char* >(malloc(2000+strlen(arg)));
```
Comment (codereport) [inline jsrc/jeload.cpp:174]:
```suggestion
```

File: jsrc/jeload.cpp [hunk 2]
```diff
@@ -221,15 +225,12 @@ int jefirst(int type,char* arg)
...
-	strcat(msg, pathdll);
-	strcat(msg," failed: ");
 
-	char ermsg[1024];
-	if(errno&&!strerror_r(errno,ermsg,1024))strcat(msg,ermsg);
-	strcat(msg,"\n");
+
 }
 
+// Refactored to C++
```
Comment (codereport) [inline jsrc/jeload.cpp:232]:
```suggestion
```

**Review by codereport — CHANGES_REQUESTED** [review]
LGTM, just comment removal

**Comment by ahmadbeirkdar** [comment]
> LGTM, just comment removal

Done!

**Comment by ahmadbeirkdar** [comment]
Getting used to the CLion git client, seems like it messed up on my end, and I pushed the wrong commits.... My bad

**Review by codereport — APPROVED** [review]
:fire:

**Pull request merged**